src/sentry/tasks/relay.py

```diff
@@ -247,6 +248,36 @@ def schedule_invalidate_project_config(
     project_id=None,
     public_key=None,
     countdown=5,
+):
+    """Enqueues :func:`_schedule_invalidate_project_config` to run after the
+    ongoing database transaction is committed. See
+    :func:`_schedule_invalidate_project_config`'s docstrings to learn how to use
+    it.
+
+    You should use this function instead of
+    :func:`_schedule_invalidate_project_config`. If not, it's possible the
+    project config invalidation job finishes before the database transaction is
+    completed, resulting in producing an stale project config that may cause a
+    production issue.
+    """
```

Comment:
I would replace this docstring with the one of `_schedule_invalidate_project_config`, and extend it by a paragraph like "When this function is called from within a db transaction, scheduling will be delayed until after the transaction has finished, to ensure that the db is up-to-date at the time of invalidation. When no database transaction is currently ongoing, the invalidation task is scheduled immediately."

Reply:
What would the docstring of `_schedule_invalidate_project_config` be? I wanted to avoid duplicating them, and it makes more sense to me to leave the parameter usage docs in the function where they are used.